stairway/src/main/java/bio/terra/stairway/impl/FlightDao.java

```diff
@@ -640,6 +640,40 @@ private FlightContextImpl resumeInner(String stairwayId, String flightId)
     }
   }
 
+  void storePersistedStateMap(String flightId, FlightMap persistedStateMap)
+      throws StairwayException, DatabaseOperationException, InterruptedException {
+    DbRetry.retryVoid(
+        "flight.storePersistedStateMap",
+        () -> storePersistedStateMapInner(flightId, persistedStateMap));
+  }
+
+  private void storePersistedStateMapInner(String flightId, FlightMap persistedStateMap)
+      throws SQLException {
+    final String sqlUpsert =
+        "INSERT INTO "
+            + FLIGHT_PERSISTED_TABLE
+            + "(flightId, key, value) VALUES (:flightId, :key, :value1) "
+            + "ON CONFLICT ON CONSTRAINT pk_flightpersisted "
+            + "DO UPDATE SET value = :value2";
+
+    List<FlightInput> inputList = FlightMapUtils.makeFlightInputList(persistedStateMap);
+
+    try (Connection connection = dataSource.getConnection();
+        NamedParameterPreparedStatement statement =
+            new NamedParameterPreparedStatement(connection, sqlUpsert)) {
+      startTransaction(connection);
+      statement.setString("flightId", flightId);
+
```

Comment:
I don't know if it's worth a short comment here, but I always forget about the fact that `NamedParameterPreparedStatement` doesn't allow reuse of named parameter.  This confused me when looking at the `ON CONFLICT...` logic, because I initially thought that `value1` and `value2` had some kind of mapping to `ProgressMeter` `v1/v2` semantics.